storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -356,4 +357,18 @@ public void onNewFinalizedCheckpoint(Checkpoint finalizedCheckpoint) {
     finalizedCheckpointChannel.onNewFinalizedCheckpoint(finalizedCheckpoint);
     forkChoiceStrategy.ifPresent(strategy -> strategy.maybePrune(finalizedCheckpoint.getRoot()));
   }
+
+  private static class SignedBlockAndStateAndSlot extends SignedBlockAndState {
```

Comment:
I wonder if we should just call this `ChainHead` or `ChainPoint` or `ChainLink` or something...  It's not quite right but I don't really like the proliferation of XandY kind of classes and would love to come up with better names for them.

Potentially we could just use this class to replace a bunch of them since the slot could just be taken from the state and then be equivalent to `SignedBlockAndState` which may be what you mean by a bigger refactoring.  Definitely happy to defer that change.